site/pages/create-consequence-stops.page.tsx

```diff
@@ -0,0 +1,308 @@
+    COOKIES_CONSEQUENCE_STOPS_ERRORS,
+} from "../constants";
+import { CreateConsequenceProps, PageState } from "../interfaces";
+import { createConsequenceStopsSchema, stopsImpactedSchema } from "../schemas/create-consequence-stops.schema";
+import { typeOfConsequenceSchema } from "../schemas/type-of-consequence.schema";
+import { flattenZodErrors, getDisplayByValue, getPageStateFromCookies } from "../utils";
+import { getStateUpdater } from "../utils/formUtils";
+
+const title = "Create Consequence Stops";
+const description = "Create Consequence Stops page for the Create Transport Disruptions Service";
+
+export interface ConsequenceStopsPageInputs extends Partial<z.infer<typeof createConsequenceStopsSchema>> {}
+export interface Stop extends z.infer<typeof stopsImpactedSchema> {}
+
+const CreateConsequenceStops = ({
+    inputs,
+    previousConsequenceInformation,
+}: CreateConsequenceProps<ConsequenceStopsPageInputs>): ReactElement => {
+    const [pageState, setPageState] = useState<PageState<Partial<ConsequenceStopsPageInputs>>>(inputs);
+    const stateUpdater = getStateUpdater(setPageState, pageState);
+    const [selected, setSelected] = useState<SingleValue<Stop>>(null);
+
+    const getOptionLabel = (e: Stop) => {
+        if (e.commonName && e.indicator && e.atcoCode) {
+            return `${e.commonName} ${e.indicator} ${e.atcoCode}`;
+        } else if (e.commonName && e.atcoCode) {
+            return `${e.commonName} ${e.atcoCode}`;
+        } else {
+            return "";
+        }
+    };
+
+    const handleChange = (value: SingleValue<Stop>) => {
+        if (
+            !pageState.inputs.stopsImpacted ||
+            pageState.inputs.stopsImpacted.filter((data) => data.id === value?.id).length === 0
+        ) {
+            addStop(value);
+        }
+        setSelected(null);
+    };
+
+    const production = process.env.NODE_ENV === "production";
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const loadOptions = async (inputValue: string, _callback: (options: Stop[]) => void) => {
+        if (inputValue && inputValue.length >= 3) {
+            const searchApiUrl = !production
+                ? "https://api.test.ref-data.dft-create-data.com/v1/stops?adminAreaCode=099"
+                : `https://api.${process.env.NODE_ENV}.ref-data.dft-create-data.com/v1/stops?adminAreaCode=099`;
+            const limit = 10;
+            const queryAdder = searchApiUrl.indexOf("?") === -1 ? "?" : "&";
+            const fetchURL = `${searchApiUrl}${queryAdder}search=${inputValue}&limit=${limit}`;
+            const res = await fetch(fetchURL, { method: "GET" });
+            const data: Stop[] = z.array(stopsImpactedSchema).parse(await res.json());
+            if (data) {
+                return data;
+            }
+        }
+        return [];
+    };
+
+    const removeStop = (e: SyntheticEvent, index: number) => {
+        e.preventDefault();
+        if (pageState.inputs.stopsImpacted) {
```

Comment:
Is stopsImpacted ever undefined or is it an empty array? The reason I ask is because [] === true

Reply:
array so I added a length check